### ISSUE TITLE
 False positive for CA1508 when a variable is assigned in a lambda expression

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn-analyzers#4997]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `'{0}' is always 'null'. Remove or refactor the condition(s) to avoid dead code.`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.201](https://dotnet.microsoft.com/download/dotnet/5.0)

No CA1508 warning should be produced:

```cs
using System;
using System.Runtime.ExceptionServices;
using System.Threading;

class C
{
    void M()
    {
        var caughtException = (ExceptionDispatchInfo?)null;

        var thread = new Thread(() =>
        {
            try
            {
                // ...
            }
            catch (Exception ex)
            {
                caughtException = ExceptionDispatchInfo.Capture(ex);
            }
        });
        thread.Start();
        thread.Join();

        // ⚠ CA1508 'caughtException' is always 'null'. Remove or refactor the condition(s) to avoid dead code.
        caughtException?.Throw();
    }
}
```

**Comment by watfordgnf** [comment]
#4984 Related? Both are in catch blocks.

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/6811
